### PR TITLE
Git storage

### PR DESCRIPTION
**Pull request by thormeier**
Add GitPersistence and some SF2 commands based on YamlPersistence to make sure the translation files are stored in a Git repository

### CODE REVIEW

**Comment by thormeier** [comment]
Rebased onto master, waiting for Travis to pass this.

**Comment by krtek4** [comment]
Seems ok to me :)

Just one thing bothers me a bit. The git commands are separated between the InitCommand and the GitPersistance, maybe put everything in GitPersistance. This would also enable creation of the repository from the application.

File: Command/GitPersistenceInitCommand.php [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+            $remote = $dialog->ask($output, 'Please enter a Git remote to clone and pull from and to push to: ');
+
+            if (!$dialog->askConfirmation($output, '<question>' . $remote . ' will be cloned into ' . $dir . '. Continue? [y] </question>', false)) {
+                return;
+            }
+        }
+
+        if (false === is_dir($dir)) {
+            $output->writeln('Creating directory "' . $dir . '"...');
+            $directoryProcess = new Process('mkdir -p ' . $dir);
```
Comment (dbu) [inline Command/GitPersistenceInitCommand.php]: can't you use the Filesystem class of symfony? it [creates a directory recursively](https://github.com/symfony/symfony/blob/2.7/src/Symfony/Component/Filesystem/Filesystem.php#L89) - and you can get rid of all the error checking as it throws meaningful exceptions.

File: Command/GitPersistenceInitCommand.php [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+        } else {
+            $output->writeln('Directory "' . $dir . '" already exists, not creating.');
+            if (is_dir($dir . DIRECTORY_SEPARATOR . '.git')) {
+                throw new \RuntimeException('"' . $dir . '" already is a git repository.');
+            }
+        }
+
+        $output->writeln('Cloning...');
+        $gitProcess = new Process(sprintf(
+            'git clone %s %s',
```
Comment (dbu) [inline Command/GitPersistenceInitCommand.php]: could move this to the GitPersistence class too.

File: Command/GitPersistenceInitCommand.php (hunk 1 above)
Comment (lsmith77) [inline Command/GitPersistenceInitCommand.php]: +1

Comment (thormeier) [inline Command/GitPersistenceInitCommand.php]: Moved to persistence in b630cd2

File: Command/GitPersistenceInitCommand.php (hunk 2 above)
Comment (thormeier) [inline Command/GitPersistenceInitCommand.php]: Moved to persistence in b630cd2

**Pull request merged**